wit/window.wit

```diff
@@ -0,0 +1,228 @@
+    use graphics-context.{graphics-context};
+    use wasi:io/poll@0.2.0.{pollable};
+    
+    
+    /// https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_code_values
+    enum key-code {
+        // TODO is this needed? I don't think the MVP needs key codes.
+        unidentified,
+    }
+    
+    /// https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#document_keys
+    /// Excerpt of the most useful keys.
+    enum special-key {
+       alt,
+       alt-graph,
+       caps-lock,
+       ctrl,
+       meta,
+       num-lock,
+       scroll-lock,
+       shift,
+       
+       arrow-down,
+       arrow-left,
+       arrow-right,
+       arrow-up,
+       end,
+       home,
+       page-down,
+       page-up,
+       
+       backspace,
+       clear,
+       delete,
+       insert,
+       paste,
+       
+       context-menu,
+       escape,
+       pause,
+       
+       brightness-down,
+       brightness-up,
+       print,
+       
+       f1,
+       f2,
+       f3,
+       f4,
+       f5,
+       f6,
+       f7,
+       f8,
+       f9,
+       f10,
+       f11,
+       f12,
+       f13,
+       f14,
+       f15,
+       f16,
+       f17,
+       f18,
+       f19,
+       f20,
+       
+       app-switch,
+       
+       channel-down,
+       channel-up,
+       fast-forward,
+       media-pause,
+       media-play,
+       media-play-pause,
+       media-record,
+       media-rewind,
+       media-stop,
+       media-track-next,
+       media-track-previous,
+       
+       audio-volume-down,
+       audio-volume-up,
+       audio-volume-mute,
+    }
+    
+    flags modifiers {
+        ctrl,
+        alt,
+        alt-graph,
+        shift,
+        caps-lock,
+        meta,
+    }
+    
+    /// Describes the key location on the keyboard, for keys that appear more than once on a keyboard.
+    enum key-location {
+        standard,
+        left,
+        right,
+        numpad,
+    }
+    
+    variant key {
+        /// The key has an associated Unicode scalar value.
+        character(u32),
+        /// The key is a key without a Unicode scalar value.
+        special(special-key),
+        /// A dead key was pressed. The next key event will contain the combined character.
+        dead,
+        /// The key could not be identified.
+        unidentified,
+    }
+    
+    
+    record key-data {
+        modifiers: modifiers,
+        key: key,
+        code: key-code,
+        location: key-location,
+        repeat: bool,
+    }
+    
+    flags mouse-buttons {
+        left,
+        middle,
+        right,
+    }
+    
+    record mouse {
+        position: position,
+        buttons: mouse-buttons,
+    }
+    
+    record scroll {
+        mouse: mouse,
+        x: f32,
+        y: f32,
+    }
+    
+    
+    /// Window system capabilities that are supported.
+    /// Methods that use unsupported capabilities will trap in WASM.
+    flags capabilities {
```

Comment:
Maybe add resize?

Reply:
Oh yeah, forgot that.

Reply:
https://github.com/WebAssembly/wasi-webgpu/commit/5162b5e046c4a118121b7e7fac5da117bb517a76#diff-b4e12b94beccf6ff7685310d97d191fb8d72eca5028b87285ca5a08a5ea44059R157